src/Authenticator/TokenAuthenticator.php

```diff
@@ -72,14 +72,14 @@ protected function stripTokenPrefix($token, $prefix)
      * Gets the token from the request headers
      *
      * @param \Psr\Http\Message\ServerRequestInterface $request The request that contains login information.
-     * @param string $headerLine Header name
+     * @param string|null $headerLine Header name
      * @return string|null
      */
     protected function getTokenFromHeader(ServerRequestInterface $request, $headerLine)
     {
-        if (!empty($headerLine)) {
+        if ($headerLine !== null) {
             $header = $request->getHeaderLine($headerLine);
-            if (!empty($header)) {
+            if ($header !== '' && $header !== null) {
```

Comment:
In general `if (!$header)` removes the cloaking and suffices. Usually way more clean then this which only forgets other nullish values.

Reply:
Can you point me to what you mean by "cloaking" so I can read something about it? Or just name me some terms https://www.google.de/search?q=php+cloaking didn't reveal anything useful to me?

Reply:
https://book.cakephp.org/3.0/en/contributing/cakephp-coding-conventions.html#careful-when-using-empty-isset

Reply:
And don't you think `if (!$var) {` will break if $var is non-boolean typehinted in php71/72 with strict mode?

Reply:
silent var not exists cloaking. if a typo is in the variable.

Reply:
Okay I understand. You mean isset($foo[$bar]) "cloaks" $foo - even if it does not exists you will see no warning or error anywhere, correct?

Good news for !$var checks as documented in the book, in php71/72: https://3v4l.org/n4iK2